core/src/jsMain/kotlin/io/islandtime/internal/Math.kt

```diff
@@ -0,0 +1,36 @@
+@file:Suppress("NewApi")
+
+package io.islandtime.internal
+
+
+//TODO replace with Math functions
+internal actual infix fun Long.floorMod(other: Long): Long = ((this % other) + other) % other
```

Comment:
I refactored the top ones to use the Kotlin APIs.
But not sure how this will work for IslandTime implementation.
please let me know if there is anyother math function that I can use here